src/pgduckdb_types.cpp

```diff
@@ -546,12 +659,33 @@ ConvertPostgresToDuckColumnType(Form_pg_attribute &attribute) {
 	case REGCLASSOID:
 		return duckdb::LogicalTypeId::UINTEGER;
 	default: {
+		if (IsEnumType(type)) {
+			return ConvertPostgresEnumToDuckEnum(type);
+		}
 		std::string name = "UnsupportedPostgresType (Oid=" + std::to_string(type) + ")";
 		return duckdb::LogicalType::USER(name);
 	}
 	}
 }
 
+Oid
+GetEnumTypeOid(const Vector &oids) {
```

Comment:
Afaict we can remove this whole function if we store the oid of the enum type also in the `PGDuckDBEnumTypeInfo` when we create it. Not needing code is always good, especially when the code causes us to cross the Postgres <-> DuckDB boundary.

Reply:
Yea agreed 👍